frontend/src/components/teacherTasks/index.js

```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import {Link, withRouter} from "react-router-dom";
+import {Box, Button, Container, Paper, Table, TableBody, TableCell, TableRow, Typography} from "@material-ui/core";
+
+class TeacherTasks extends React.Component {
+
+    constructor(props) {
+        super(props);
+
+        this.handleRowSelect = this.handleRowSelect.bind(this);
+    }
+
+    handleRowSelect(event) {
+        console.log(JSON.stringify(this.props))
+        // TODO: Send extra information about the task to the teacherTaskViewer.
+        this.props.history.push("/teacherTaskViewer")
+    }
```

Comment:
This is one solution but i think a better and the more versatile would be to try and do this state sharing with react hooks
https://www.codementor.io/@sambhavgore/an-example-use-context-and-hooks-to-share-state-between-different-components-sgop6lnrd